Check for instance termination using autoscaling/target-lifecycle-state

wbecker opened the pull request:
Fixes #1

lexicallabs committed:
Check for instance termination using autoscaling/target-lifecycle-state

Fixes #1

brendtumi commented on README.md:
```diff
@@ -22,6 +22,9 @@ spot.once("termination-cancelled", date => {
     console.log("termination-not-detected", error);
 });
+spot.on("instance-termination", error => {
+    console.log("termination-not-detected", error);
```
can you update the text as "instance-termination" in example.

wbecker replied:
Good spot :)

brendtumi reviewed (changes requested):
Thank you very much, if you can update the example we can merge it.

wbecker commented:
Not 100% sure this is working to be honest - had this running on an instance overnight and it didn't seem to pick it up. It seems like it should be working so it could just be my code that is using it! I'll test it for another day or two then when I'm happy I'll poke you and ask you to merge!

brendtumi commented:
> Not 100% sure this is working to be honest - had this running on an instance overnight and it didn't seem to pick it up. It seems like it should be working so it could just be my code that is using it! I'll test it for another day or two then when I'm happy I'll poke you and ask you to merge!

Thanks, I will wait until you got a successful result.

lexicallabs committed:
Handle termination-time endpoint 404-ing when no data.

This fires an error that we don't want to handle as a rejection,
instead return null and handle this case.

lexicallabs committed:
Correct bad log in readme

wbecker commented:
It had been failing on my server because the new code was trying to get the results from 2 separate endpoints. Previously, when not in a terminating state the code would get a 404 from termination-time endpoint and error silently. The code in this PR now gets data from two endpoints, so it can't just throw if one of the endpoint 404s. 

I now handle this condition and it seems like it is working on my server now. I'll let it run for 24 hours and make sure there are no further issues, then let's merge!

wbecker commented:
![image](https://user-images.githubusercontent.com/348044/181253346-3b25075c-5059-4f85-8aa5-e249c83f9889.png)

Seems good now - you can see the temporary logging I've added on my staging server.

brendtumi commented:
version 2.1.0 is published, thank you for your contribution.
